offload/liboffload/API/CMakeLists.txt

```
@@ -13,15 +13,31 @@ if (CLANG_FORMAT)
     tablegen(OFFLOAD OffloadPrint.hpp -gen-print-header)
     tablegen(OFFLOAD OffloadErrcodes.inc -gen-errcodes)
 
-    set(FILES_TO_COPY "OffloadAPI.h;OffloadEntryPoints.inc;OffloadFuncs.inc;OffloadImplFuncDecls.inc;OffloadPrint.hpp")
-    set(GEN_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../include/generated)
+    set(files_to_copy "OffloadAPI.h;OffloadEntryPoints.inc;OffloadFuncs.inc;OffloadImplFuncDecls.inc;OffloadPrint.hpp")
+    set(generated_dir ${CMAKE_CURRENT_SOURCE_DIR}/../include/generated)
+    set(shared_dir ${LIBOMPTARGET_INCLUDE_DIR}/Shared)
     add_public_tablegen_target(OffloadGenerate)
-    add_custom_command(TARGET OffloadGenerate POST_BUILD COMMAND ${CLANG_FORMAT}
-        -i ${TABLEGEN_OUTPUT})
-    add_custom_command(TARGET OffloadGenerate POST_BUILD COMMAND ${CMAKE_COMMAND}
-        -E copy_if_different ${FILES_TO_COPY} ${GEN_DIR})
-    add_custom_command(TARGET OffloadGenerate POST_BUILD COMMAND ${CMAKE_COMMAND}
-        -E copy_if_different OffloadErrcodes.inc "${LIBOMPTARGET_INCLUDE_DIR}/Shared/OffloadErrcodes.inc")
+
+    add_custom_target(OffloadAPI DEPENDS OffloadGenerate)
```

```suggestion
    add_custom_target(OffloadAPI)
    add_dependencies(OffloadAPI OffloadGenerate)
```
`DEPENDS` is for file-level dependencies